rust-wasm-id-allocator/distributed-id-allocator/src/compressor/tables/session_space.rs

```diff
@@ -187,7 +185,7 @@ impl SessionSpace {
     pub fn new(session_id: SessionId) -> SessionSpace {
         SessionSpace {
             session_id,
-            cluster_chain: Vec::new(),
+            cluster_chain: Vec::with_capacity(1),
```

Comment:
FYI this significantly improves memory pressure for normal scenarios, as the default initial size after first push is 4 currently :(

Reply:
0, it's lazy if you don’t specify.

Reply:
Oh I understand. You mean that when we do our first push into the Vec, its resulting size is then 4. So in a degenerate case of a large number of single-ID clusters, we'd be quadrupling memory space. Yes?

Reply:
Well, the capacity/count doesn't matter. It's how many clusters the session made, which on average is probably 1-2. I could see an argument for making this 2, but for now 1 is better than 4 :P